src/common/AndroidUtils.ts

```diff
@@ -154,157 +139,147 @@ export class AndroidSDKUtils {
-            }
+            return CommonUtils.executeCommandAsync(
+                `${AndroidSDKUtils.getSdkManagerCommand()} --list`
+            )
+                .then((result) => {
+                    if (result.stdout && result.stdout.length > 0) {
+                        const packages = AndroidPackages.parseRawPackagesString(
+                            result.stdout
+                        );
+                        AndroidSDKUtils.packageCache = packages;
+                    }
+                    return Promise.resolve(AndroidSDKUtils.packageCache);
+                })
+                .catch((error) => Promise.reject(error));
+        } else {
+            return Promise.resolve(AndroidSDKUtils.packageCache);
         }
-
-        return Promise.resolve(AndroidSDKUtils.packageCache);
     }
 
-    public static fetchEmulators(): Promise<AndroidVirtualDevice[]> {
-        return new Promise((resolve, reject) => {
-            let devices: AndroidVirtualDevice[] = [];
-            try {
-                const result = CommonUtils.executeCommandSync(
-                    AndroidSDKUtils.getAvdManagerCommand() + ' list avd'
-                );
-                if (result) {
+    public static async fetchEmulators(): Promise<AndroidVirtualDevice[]> {
+        let devices: AndroidVirtualDevice[] = [];
+        return CommonUtils.executeCommandAsync(
+            AndroidSDKUtils.getAvdManagerCommand() + ' list avd'
+        )
+            .then((result) => {
+                if (result.stdout && result.stdout.length > 0) {
                     devices = AndroidVirtualDevice.parseRawString(
-                        result.toString()
+                        result.stdout
                     );
                 }
-            } catch (exception) {
+                return Promise.resolve(devices);
+            })
+            .catch((exception) => {
                 AndroidSDKUtils.logger.warn(exception);
-            }
-            return resolve(devices);
-        });
+                return Promise.resolve(devices);
+            });
     }
 
     public static async findRequiredAndroidAPIPackage(): Promise<
         AndroidPackage
     > {
-        try {
-            const minSupportedRuntimeAndroid = Version.from(
-                androidConfig.minSupportedRuntimeAndroid
-            );
+        const minSupportedRuntimeAndroid = Version.from(
+            androidConfig.minSupportedRuntimeAndroid
+        );
 
-            const packages = await AndroidSDKUtils.fetchInstalledPackages();
-            if (packages.isEmpty()) {
-                return Promise.reject(
-                    new Error(
-                        `No Android API packages are installed. Minimum supported Android API package version is ${androidConfig.minSupportedRuntimeAndroid}`
-                    )
-                );
-            }
+        return AndroidSDKUtils.fetchInstalledPackages()
+            .then(async (packages) => {
```

Comment:
What does `async` buy you here?

Reply:
on line 214 I'm using `await` and you can't do that without `async` here.